src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java

```diff
@@ -650,9 +682,169 @@ private void fireNdefEvent(String type, Ndef ndef, Parcelable[] messages) {
+    private void fireMifareUltralightEvent(Tag tag)
+    {
+        JSONObject tagJSON = Util.tagToJSON(tag);
+
+        MifareUltralight mifare = MifareUltralight.get(tag);
+        int type = mifare.getType();
+        try
+        {
+            tagJSON.put("type",type);
+        }
+        catch (JSONException e)
+        {
+            e.printStackTrace();
+        }
+
+        byte[] buffer = new byte[16];
+        int sector = -1;
+        int wptr = 0;
+        try
+        {
+            mifare.connect();
+            int page = 0;
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            while((buffer = mifare.readPages(page++))!=null && buffer.length > 0)
+            {
+                out.write(buffer);
+            }
+            buffer = out.toByteArray();
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+        finally
+        {
+            try
+            {
+                mifare.close();
+            }
+            catch (IOException e1) {}
+        }
+
+        try
+        {
+            tagJSON.put("data",Util.byteArrayToJSON(buffer));
+        }
+        catch (JSONException e)
+        {
+            e.printStackTrace();
+        }
+
+        String command = MessageFormat.format(javaScriptEventTemplate, MIFARE_ULTRALIGHT, tagJSON);
+        Log.v(TAG, command);
+        this.webView.sendJavascript(command);
 
-        String command = MessageFormat.format(javaScriptEventTemplate, NDEF_FORMATABLE, Util.tagToJSON(tag));
+    }
+
+    private void fireMifareClassicEvent(Tag tag)
+    {
+        CordovaWebView webview = this.webView;
+        new AsyncTask<Tag,Void,String>()
+        {
+            protected String doInBackground(Tag... tags)
+            {
+                int sector = -1;
+                Tag tag = tags[0];
+                JSONObject tagJSON = Util.tagToJSON(tag);
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+                MifareClassic mifare = MifareClassic.get(tag);
+                int size = mifare.getSize();
+                int type = mifare.getType();
+                try
+                {
+                    tagJSON.put("type",type);
+                    tagJSON.put("size",size);
+                }
+                catch (JSONException e)
+                {
+                    e.printStackTrace();
+                }
+
+                try
+                {
+                    mifare.connect();
+                    mifare.setTimeout(3000);
+
+                    for(int i = 0; i < mifare.getBlockCount(); ++i)
+                    {
+                        if(mifare.blockToSector(i) != sector)
+                        {
+                            sector = mifare.blockToSector(i);
+                            mifare.authenticateSectorWithKeyA(sector,mifare.KEY_DEFAULT);
+                            //mifare.authenticateSectorWithKeyB(sector,mifare.KEY_DEFAULT);
```

Comment:
what's the reason for this being commented out? either keep, remove or add a comment on why it is commented out